pulp_2to3_migration/app/models/base.py

```diff
@@ -11,24 +17,113 @@ class MigrationPlan(Model):
         plan (models.JSONField): The migration plan in the JSON format
     """
     plan = JSONField()
+    _real_plan = None
+
+    @property
+    def plan_view(self):
+        """
+        Window to view the validated migration plan data through.
+        """
+        # Needs a MongoDB connection initialized beforehand to look for importers,
+        # verify existance of repositories and distributors, etc.
+        if not self._real_plan:
+            self._real_plan = _ValidatedMigrationPlan(self.plan)
+
+        return self._real_plan
+
+    def get_plugins(self):
+        """
+        Return a list of pulp2 plugins to migrate.
+        """
+        return self.plan_view.plugins_to_migrate
 
     def get_repositories(self):
         """
         Return a list of pulp2 repositories to migrate or empty list if all should be migrated.
         """
-        # TODO: get pulp2 repositories from the self.plan
-        return []
+        return self.plan_view.repositories_to_migrate
 
     def get_importers(self):
         """
         Return a list of pulp2 importers to migrate or empty list if all should be migrated.
         """
-        # TODO: get pulp2 importers from the self.plan
-        return []
+        return self.plan_view.importers_to_migrate
 
     def get_distributors(self):
         """
         Return a list of pulp2 distributors to migrate or empty list if all should be migrated.
         """
-        # TODO: get pulp2 distributors from the self.plan
-        return []
+        return self.plan_view.distributors_to_migrate
+
+
+class SchemaValidationFailure(Exception):
+    """
+    Exception to be thrown when validating the MigrationPlan.
+
+    e.g. Repository specified does not exist.
+    """
+    pass
+
+
+class _ValidatedMigrationPlan:
+    def __init__(self, migration_plan):
+        self.migration_plan = migration_plan
+
+        self.plugins_to_migrate = []
+        self.importers_to_migrate = []
+        self.distributors_to_migrate = []
+        self.repositories_to_migrate = []
+
+        self.validate_and_populate()
+
+    def validate_and_populate(self):
+        for plugin_data in self.migration_plan['plugins']:
+            self.plugins_to_migrate.append(plugin_data['type'])
+            if plugin_data.get('repositories'):
```

Comment:
@goosemania given that fact that we changed our migration strategy( i.e. we cannot migrate only the content because we need data from repo/importers),  i think we need to make repository as required field on the schema of the MP

Reply:
@ipanova, either this or in the past we agreed that we just skip repo/importer parts and migrate only downloaded content. I'l create a task  which will either change a requirement or skip unnecessary steps.

Reply:
@goosemania Please link me the task when you create it

Reply:
@dralley https://pulp.plan.io/issues/5549